plugin/dnstap/handler.go

```diff
@@ -66,20 +71,31 @@ func (h Dnstap) tapQuery(ctx context.Context, w dns.ResponseWriter, query *dns.M
 
 // ServeDNS logs the client query and response to dnstap and passes the dnstap Context.
 func (h Dnstap) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
-	rw := &ResponseWriter{
-		ResponseWriter: w,
-		Dnstap:         h,
-		query:          r,
-		ctx:            ctx,
-		queryTime:      time.Now(),
+	qt := time.Now()
```

Comment:
if we are filtering.. should we only do time.Now inside the ifs? that would reduce overhead when we know it won't be sent